datafusion/physical-expr/src/expressions/case.rs

```diff
@@ -286,6 +290,89 @@ impl PhysicalExpr for CaseExpr {
             self.case_when_no_expr(batch)
         }
     }
+
+    fn children(&self) -> Vec<Arc<dyn PhysicalExpr>> {
+        let mut chileren = vec![];
+        match &self.expr {
+            Some(expr) => chileren.push(expr.clone()),
+            None => chileren.push(Arc::new(NoOp::new())),
+        }
+        self.when_then_expr.iter().for_each(|(cond, value)| {
+            chileren.push(cond.clone());
+            chileren.push(value.clone());
+        });
+
+        match &self.else_expr {
+            Some(expr) => chileren.push(expr.clone()),
+            None => chileren.push(Arc::new(NoOp::new())),
+        }
+        chileren
+    }
+
+    // For physical CaseExpr, we do not allow modifying children size
+    fn with_new_children(
+        self: Arc<Self>,
+        children: Vec<Arc<dyn PhysicalExpr>>,
+    ) -> Result<Arc<dyn PhysicalExpr>> {
+        if children.len() != self.children().len() {
+            Err(DataFusionError::Internal(
+                "CaseExpr: Wrong number of children".to_string(),
+            ))
+        } else {
+            assert_eq!(children.len() % 2, 0);
+            let expr = match children[0].clone().as_any().downcast_ref::<NoOp>() {
+                Some(_) => None,
+                _ => Some(children[0].clone()),
+            };
+            let else_expr = match children[children.len() - 1]
```

Comment:
i would find it easier to read if the `else_expr` handling was after the `when_then_expr` handling to mirror the `children()` but I think this is fine too